### ISSUE TITLE
get_trend_data method doesn't return anything

### ISSUE THREAD

**Issue by cholmescanada** [scottbonline/sense#58]
https://github.com/scottbonline/sense/blob/9b46e7a39342b55c3ff0ade24a0571f1aa30c577/sense_energy/senseable.py#L192

I am not able to get any data from sense.get_trend_data('DAY') even after running sense.update_trend_data()

I am able to authenticate and have other device usage data returned such as sense.daily_usage

Apologies if this is just an error on my end - would appreciate any help. 

Thanks!

**Comment by kbickar** [comment]
The data resets at midnight, maybe it works for you now?
You can dump it all to see with `print(sense._trend_data["DAY"])`

**Comment by cholmescanada** [comment]
Thanks for the reply! When I try that, I do get an output. 

I am sure this is a problem with my understanding, however what I am looking to do is get usage data on a certain date. I see in the code it allows specifying a date, however I am unable to get anything to return.

`import datetime
from sense_energy import Senseable
sense = Senseable()
sense.authenticate('<user>', '<pass>')

my_date = datetime.datetime(2023, 4, 2)
test = sense.get_trend_data('DAY',my_date)
print(test)`

and I get "None" as the output.

Thanks in advance!

**Comment by sagararthonsys** [comment]
I am not able to get any data from sense.get_trend_data('DAY') even after running sense.update_trend_data()

**Comment by dauc** [comment]
`sense.get_trend_data` doesn't actually return the trend data; it just retrieves it and stores it in `sense._trend_data`.

Based on the other `get_` functions maybe it should `return self._trend_data[scale]` ?  But it works to call and then read the result from `self._trend_data[scale]` directly.

**Comment by snairolf** [comment]
Agree, the naming is inconsistent. Based on the other functions get_trend_data() should just do the api call and not change the internal _trend_data. 

Instead, update_trend_data could have a scales=[] input parameter, or similar.

**Comment by kbickar** [comment]
I guess `get_trend_data()` should really be an internal function or renamed to `update_single_trend_data()` or something.  The function `get_trend()` should be used to retrieve trend data after it's been updated

**Comment by snairolf** [comment]
Renaming to `update_single_trend_data()` would make sense. It would still be good to keep `get_trend_data()` in line with all the other functions that just return the raw data from the API call. 

Currently, it's quite mixed which functions return values AND modify internal data versus the ones that just return the raw data from the call (modifying internal data: `get_trend_data(), get_discovered_device_names(), get_monitor_data()`). IMHO, it would be good to have a better separation in terms of naming between the functions that do an API call vs the ones that don't.

**Comment by kbickar** [comment]
The reason for that is the `get_discovered_device_names()` and `get_monitor_data()` are API calls that would likely be called only once per session while the ones like `get_trend()` would return the data from the last time it was updated (which might happen once a minute or more).   The `get_trend()` could be called quite frequently while there's not much reason to call the others more than once

Renaming that function and a doc update would work here

**Comment by jay3702** [comment]
I found that this works:
`import datetime
from sense_energy import Senseable
from sense_energy import Scale
from datetime import datetime
from datetime import timezone


def main():
    # Replace these with your actual Sense credentials.
    sense = Senseable()
    sense.authenticate(username, password)
    sense.update_realtime()
    sense.update_trend_data()
    print ("Active:", sense.active_power, "W")
    print ("Active Solar:", sense.active_solar_power, "W")
    print ("Daily:", sense.daily_usage, "KWh")
    print ("Daily Solar:", sense.daily_production, "KWh")
    print ("Active Devices:",", ".join(sense.active_devices))
    date_str = "01/01/2023"
    date_obj = datetime(2023, 1, 1, 8, 0, 0, tzinfo=timezone.utc)
    sense.get_trend_data(Scale.YEAR, date_obj)
    print(sense._trend_data[Scale.YEAR])`

The trend data for all time scales are in sense._trend_data

-Jay

**Comment by kbickar** [comment]
You shouldn't need to access internal class variables directly, just call `get_stat(Scale.YEAR, key)` - this returns the same data with error checking included

**Comment by jay3702** [comment]
Cool! I have used the export feature of the Sense web app which creates a file for download. I am currently parsing out the mains and solar values to create a daily net energy report which I use to estimate the actual cost of charging my ev's adjusted for solar production. I think the data I need is in the trend data object. It seems that the get_stat function returns a single value. Is there a way to get a dictionary of values?

**Comment by kbickar** [comment]
There isn't - you need to specify which key you want
